### ISSUE TITLE
Fatal error (on rate limiting?)

### ISSUE THREAD

**Issue by tjconcept** [podio/podio-js#51]
I'm really sorry I can't describe this bug in more detail. It happens consistently after some requests so I assume it's either time dependent or rate limiting.

```sh
[..]/node_modules/podio-js/lib/transport.js:111
      var isInvalidOrExpiredToken = (res.body.error === 'invalid_token' || res.body.error_description === 'expired_token');
                                        ^

TypeError: Cannot read property 'body' of undefined
    at PodioJS._onResponse ([..]/node_modules/podio-js/lib/transport.js:111:41)
    at Request.callback ([..]/node_modules/superagent/lib/node/index.js:815:12)
    at ClientRequest.<anonymous> ([..]/node_modules/superagent/lib/node/index.js:767:10)
    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at TLSSocket.socketErrorListener (_http_client.js:399:9)
    at emitOne (events.js:115:13)
    at TLSSocket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:62:8)
    at _combinedTickCallback (internal/process/next_tick.js:102:11)
    at process._tickCallback (internal/process/next_tick.js:161:9)
```

**Comment by corsacca** [comment]
I just got this too.

**Comment by Andrianina007** [comment]
how many rate limit you have ( may be you can find it on res.headers)

**Comment by corsacca** [comment]
res.headers is not available. The client library does not return res

**Comment by dmatteo** [comment]
Hi @tjconcept,

you might be hitting the rate-limit assigned to your API key.

You can monitor your current status by examining two HTTP headers that are returned with each request you make:

```
X-Rate-Limit-Limit: The ceiling for the request you just made
X-Rate-Limit-Remaining: The number of requests you have left for the current 1 hour window
```

Source: https://developers.podio.com/index/limits

**Comment by tjconcept** [comment]
Hi @dmatteo,

Wouldn't that be a `PodioRateLimitError` error then?

Whatever reason, I guess it's still a bug in this code base, unless that is the documented behaviour???

**Comment by corsacca** [comment]
@dmatteo How do we access those headers using this podio-js library?

**Comment by corsacca** [comment]
@dmatteo I'm not hitting the rate limit, but i'm still getting the same error. I don't know why res is not defined at this point.

**Comment by cmchase** [comment]
We've patched a small fix for this blocking error into version [1.6.1](https://www.npmjs.com/package/podio-js)
